apps/server/src/modules/room/domain/service/room-content.service.ts

```diff
@@ -0,0 +1,63 @@
+import { BadRequestException, Injectable } from '@nestjs/common';
+import { EntityId } from '@shared/domain/types';
+import { RoomContentRepo } from '../../repo';
+import { RoomContentItem, RoomContentType } from '../type';
+
+@Injectable()
+export class RoomContentService {
+	constructor(private readonly roomContentRepo: RoomContentRepo) {}
+
+	public async contentExists(roomId: EntityId): Promise<boolean> {
+		const exists = (await this.roomContentRepo.countByRoomId(roomId)) > 0;
+
+		return exists;
+	}
+
+	public async createContent(roomId: EntityId, boardIds: EntityId[]): Promise<void> {
+		const items: RoomContentItem[] = boardIds.map((boardId) => {
+			return { type: RoomContentType.BOARD, id: boardId };
+		});
+		await this.roomContentRepo.createContent(roomId, items);
+	}
+
+	public async getBoardOrder(roomId: EntityId): Promise<EntityId[]> {
+		const items = await this.roomContentRepo.findContentItemsByRoomId(roomId);
+		const boardIds = items.filter((item) => item.type === RoomContentType.BOARD).map((item) => item.id);
+
+		return boardIds;
+	}
+
+	public async addBoard(roomId: EntityId, boardId: EntityId): Promise<void> {
```

Comment:
consider rename to something like addContentReference
ideally, the type is also given, but maybe should we have a check if reference content actually exists, or this is resposability of the caller?

also, the id should not already exist in the array (avoid adding it twice)?